service/paxos/paxos_state.cc

```diff
@@ -81,73 +81,6 @@ static dht::shard_replica_set shards_for_writes(const schema& s, dht::token toke
     return shards;
 }
 
-future<paxos_state> paxos_state::load_and_repair_paxos_state(db::system_keyspace& sys_ks,
```

Comment:
> Any paxos state read which drives some paxos round forward is already
accompanied by a paxos state write. 

I guess this is the crux. If every read is also a write (to the same paxos row), then an obervation of the old state will heal the old+new state by overwriting it.

But it is always true? Can't we sometimes read and return failure (without writing anything new)?

Reply:
replied in https://github.com/scylladb/scylladb/pull/24926#issuecomment-3068522509

Reply:
yes, but #24926 is ready for merge